### ISSUE TITLE
Globally injected variables do not work in subcharts/dependencies

### ISSUE THREAD

**Issue by dvvanessastoiber** [opendevstack/ods-jenkins-shared-library#1066]
**Describe the bug**
In the MRO pipeline, Jenkins sets the parameter imageTag during the helm install. If the deployment.yaml is located in a dependent chart (charts/ folder) and contains `image: "{{ .Values.image.repository }}:{{ .Values.imageTag }}"`, helm cannot overwrite imageTag. This can only be achieved by accessing and setting the global parameter `image: "{{ .Values.image.repository }}:{{ .Values.global.imageTag }}"`; `--set global.imageTag=x.x.x`

**To Reproduce**
1. Clone Helm Chart Repo with nested chart structure
2. cd chart/
3. Access .Values.imageTag in deployment.yaml
4. Run `helm template .  -f values.yaml -f values.dev.yaml -f secrets.dev.yaml --set registry=image-registry.openshift-image-registry.svc:5000 --set componentId=<componentId> --set imageNamespace=<project>-dev --set imageTag=abc`
5. Check imageTag used for deployment
6. Use .Values.global.imageTag in deployment.yaml
7. Run `helm template .  -f values.yaml -f values.dev.yaml -f secrets.dev.yaml --set registry=image-registry.openshift-image-registry.svc:5000 --set componentId=<componentId> --set imageNamespace=<project>-dev --set global.imageTag=abc`
8. Check imageTag used for deployment

**Expected behavior**
imageTag should be set by MRO pipeline

**Screenshots**
-

**Affected version:**
 - OpenDevStack v4.x

**Comment by michaelsauter** [comment]
FYI we actually set `global.imageTag` already for a while for exactly the same use case :) would be nice to have it out of the box!

**Comment by dvvanessastoiber** [comment]
> @dvvanessastoiber did you get a chance to give this a try?

@serverhorror Thank you for providing this workaround! I tested it and have a few open questions:

Prerequisites: We edited our base helm chart that now uses the global imageTag if it is set.

To my understanding, the DEV environment does not have any generated tags in the ImageStream and the actual version of the application image can be used. In contrast to that, QA and PROD only have ODS generated tags in the ImageStream (see screenshot below).

![image](https://github.com/opendevstack/ods-jenkins-shared-library/assets/52395160/f917a460-cdb6-4525-880e-069446469ca0)

1. With the provided workaround in the Jenkinsfile, I cannot distinguish between stages, therefore the global value is set for dev as well, even if there is no need for a generated tag in this stage (please correct me if I am wrong)
2. The context.shortGitCommit is entirely different from the ODS generated tag (see screenshot above).

Please let me know what you think regarding those points! Thank you!

**Comment by dvvanessastoiber** [comment]
> > @dvvanessastoiber did you get a chance to give this a try?
> 
> @serverhorror Thank you for providing this workaround and apologies for the delayed response! I have tested it and have a few open questions:
> 
> Prerequisites: We edited our base helm chart that now uses the global imageTag if it is set.
> 
> To my understanding, the DEV environment does not have any generated tags in the ImageStream and the actual version of the application image can be used. In contrast to that, QA and PROD only have ODS generated tags in the ImageStream (see screenshot below).
> 
> ![image](https://private-user-images.githubusercontent.com/52395160/304099249-f917a460-cdb6-4525-880e-069446469ca0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgzNTgxMjMsIm5iZiI6MTcwODM1NzgyMywicGF0aCI6Ii81MjM5NTE2MC8zMDQwOTkyNDktZjkxN2E0NjAtY2RiNi00NTI1LTg4MGUtMDY5NDQ2NDY5Y2EwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE5VDE1NTAyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk4ZjEwYzNhMzBiMjNkNWZmNmE4MDA3YTIwNmE5MTkwNjkzZjQwNzc2NGRhZmY5MDE0MGUwZjZiZWViYjc2NTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.fBlpqtZ2EoaV6vgvsQvNCIiFqJSWdaLSR0YS1-k-aOM)
> 
> 1. With the provided workaround in the Jenkinsfile, it is not possible to distinguish between stages, therefore the global value is set for dev as well, even if there is no need for a generated tag in this stage (please correct me if I am wrong)
> 2. The context.shortGitCommit is entirely different from the ODS generated tag (see screenshot above).
> 
> Please let me know what you think regarding those points and kindly let me know if we should have a call to discuss this in detail! Thank you!

@serverhorror Did you have time to look into this yet? Thank you for your support!
